homework7/hw2.py

```diff
@@ -0,0 +1,56 @@
+"""
+Given two strings. Return if they are equal when both are typed into
+empty text editors. # means a backspace character.
+
+Note that after backspacing an empty text, the text will continue empty.
+
+Examples:
+    Input: s = "ab#c", t = "ad#c"
+    Output: True
+    # Both s and t become "ac".
+
+    Input: s = "a##c", t = "#a#c"
+    Output: True
+    Explanation: Both s and t become "c".
+
+    Input: a = "a#c", t = "b"
+    Output: False
+    Explanation: s becomes "c" while t becomes "b".
+
+"""
+
+
+def backspace_compare(first: str, second: str) -> bool:
+    """
+    Function that compare two str with symbol "#" means a backspace
+    param:
+        first (str): input string
+        second(str) : input string
+    return: True is strings are equal and False if not
+    """
+    if str_without_backspace(first) == str_without_backspace(second):
+        return True
```

Comment:
when you do conditions with simple "return True"/"return False" in condition branches you may replace by by simple "return {condition}"